centaur/src/main/scala/centaur/test/formulas/TestFormulas.scala

```diff
@@ -41,18 +41,20 @@ object TestFormulas extends StrictLogging {
   def runSuccessfulWorkflowAndVerifyMetadata(workflowDefinition: Workflow)(implicit cromwellTracker: Option[CromwellTracker]): Test[SubmitResponse] = for {
     _ <- checkDescription(workflowDefinition, validityExpectation = Option(true))
     submittedWorkflow <- runSuccessfulWorkflow(workflowDefinition)
-    metadata <- validateMetadata(submittedWorkflow, workflowDefinition)
-    _ <- validateJobManagerStyleMetadata(submittedWorkflow, metadata.value)
-    _ = cromwellTracker.track(metadata)
-    _ <- validateDirectoryContentsCounts(workflowDefinition, submittedWorkflow, metadata)
+    notArchivedMetadata <- validateMetadata(submittedWorkflow, workflowDefinition, validateArchived = Option(false))
+    _ <- validateJobManagerStyleMetadata(submittedWorkflow, originalMetadata = notArchivedMetadata.value, validateArchived = Option(false))
+    _ = cromwellTracker.track(notArchivedMetadata)
+    _ <- validateDirectoryContentsCounts(workflowDefinition, submittedWorkflow, notArchivedMetadata)
     _ <- waitForArchive(submittedWorkflow, workflowDefinition)
-    // Re-validate the metadata now that carboniting has completed
-    _ <- validateMetadata(submittedWorkflow, workflowDefinition)
-    flatMetadata = metadata.asFlat
-    workflowRoot = flatMetadata.value.get("workflowRoot").collectFirst { case JsString(r) => r } getOrElse "No Workflow Root"
+    // Compare archived and unarchived metadata in various querying scenarios
+    _ <- validateMetadataBeforeAndAfterArchival(submittedWorkflow, workflowDefinition, expandSubworkflows = false)
+    _ <- validateMetadataBeforeAndAfterArchival(submittedWorkflow, workflowDefinition, expandSubworkflows = true)
+    _ <- validateJobManagerStyleMetadataBeforeAndAfterArchival(submittedWorkflow)
```

Comment:
Should these function names be more like `validateMetadataIsTheSameBeforeAndAfterArchival `?

OTOH that name is starting to get really, really clunky...